playground/cdk/CdkSample.AppHost/Program.cs

```diff
@@ -26,9 +27,15 @@
 var sqldb = builder.AddSqlServer("sql").AsAzureSqlDatabase().AddDatabase("sqldb");
 
 var signaturesecret = builder.AddParameter("signaturesecret");
-var keyvault = builder.AddAzureKeyVault("mykv", (_, construct, keyVault) =>
+var keyvault = builder.AddAzureKeyVault("mykv", (def) =>
 {
-    var secret = new KeyVaultSecret(construct, name: "mysecret");
+    var secret = new KeyVaultSecret(def.Construct, def.KeyVault, name: "mysecret");
+    secret.AssignProperty(x => x.Properties.Value, signaturesecret);
+});
+
+var secureKeyVault = builder.AddAzureKeyVault<MoreSecureAzureKeyVaultAzureKeyVaultDefinition>("securekv", (def) =>
```

Comment:
This invocation of AddAzureKeyVault has a different definition type. The definition is responsible for controlling the shape of the resources as they get defined. It is a generic overload so that any custom properties defined can be got at (there are one in this simple exammple).